pkgs/development/ruby-modules/bundled-common/default.nix

```diff
@@ -118,7 +118,7 @@ let
 
       wrappedRuby = stdenv.mkDerivation {
         name = "wrapped-ruby-${pname'}";
-        nativeBuildInputs = [ makeWrapper ];
+        nativeBuildInputs = [ makeBinaryWrapper ];
```

Comment:
```suggestion
        nativeBuildInputs = if stdenv.isDarwin [ makeBinaryWrapper ] else [ makeWrapper ];
```

I don't think we should convert all wrappers to binary ones just because Darwin is to limited to support basic things.

Reply:
I think BSD-systems also have the same shebang issue as macOS, without having a system readily available to test this on. 

By branching, if someone using Linux adds a `--run`-flag (or another flag which is supported by makeWrapper but not makeBinaryWrapper), they could break the wrappers on macOS without noticing and without breaking them on Linux.

By using the same wrapper type on both macOS and Linux, it means a change that causes a breakage on one platform will likely do so on other platforms as well - decreasing the chance that someone using Linux will break a package for macOS without noticing.

Reply:
> While I think that we should be cautious about this kinda of change to avoid breakage, I still think this has benefits for Linux systems too. For example, reducing the dependency tree to bootstrap (see https://github.com/NixOS/nixpkgs/issues/160323, where a failure to build bash on Android is causing freetype build to fail)

No, bash didn't fail to build there. It is most likely a splicing problem that a binary for a another platform is used and this shouldn't be changed by this PR. Also I think it is safe to assume that we cannot get bash out of bootstraping something. There is always some bash somewhere involved. 

> possible binaries that starts faster (see https://github.com/NixOS/nixpkgs/pull/124556#discussion_r766199220 and https://github.com/NixOS/nixpkgs/pull/124556#discussion_r766230653)

The performance improvement would be around ~3ms. Wrapping is very widely used and debugging failed shebangs is a pain to debug. I don't think the binary wrapper is so widely tested that it is safe to assume that it just works for everything. Though it might be safe to assume that it works for ruby things.

Reply:
> By branching here on platform, if someone using Linux adds a `--run`-flag (or another flag which is supported by makeWrapper but not makeBinaryWrapper), they could break the wrappers on macOS without noticing and without breaking them on Linux.

Good point but that also means that we can't use the wrapper for everything yet.
----
Also this PR rebuilds more things than cewl.

Reply:
I don't think it's such a pain to debug such wrappers. Here with Neovim, I opened for example `./results/cewl/bin/cewl` and opened the binary wrapper interpreter from the shebang, and saw immediately the text in the binary contents:

```
# ------------------------------------------------------------------------------------
# The C-code for this binary wrapper has been generated using the following command:


makeCWrapper /nix/store/vchcc0l18dj4pk07jv1z4qhd27lq0ni1-ruby-2.7.5/bin/ruby \
    --set 'BUNDLE_GEMFILE' '/nix/store/9mbvhw5w5zm81ay0jf2jyx59g7a72zcr-gemfile-and-lockfile/Gemfile' \
    --unset 'BUNDLE_PATH' \
    --set 'BUNDLE_FROZEN' '1' \
    --set 'GEM_HOME' '/nix/store/as2jwqvgcginmn13qzdqh5c8dyc27i6h-cewl-ruby-env/lib/ruby/gems/2.7.0' \
    --set 'GEM_PATH' '/nix/store/as2jwqvgcginmn13qzdqh5c8dyc27i6h-cewl-ruby-env/lib/ruby/gems/2.7.0'


# (Use `nix-shell -p makeBinaryWrapper` to get access to makeCWrapper in your shell)
# ------------------------------------------------------------------------------------
```

So I don't feel in any way that this effects my wrappers debugging workflow, assuming I count on `makeBinaryWrapper` that this environment is actually set before the wrapper executes the final binary, and for that we have `makeBinaryWrapper`'s golden tests.

Reply:
> The performance improvement would be around ~3ms.

Someone in a group that I participate confirmed this numbers on NixOS. A quick calculation here:

- 3ms * 1000 = 3s
- 3ms * 10000 = 30s
- 3ms * 100000 = 300s

I can definitively see some script calling a program multiple times in a loop hitting this number of calls, and eventually this becomes a very large overhead.

Yeah, maybe you could argue that a script calling a program this number of times is broken, and it should be converted to a program or etc, however I would argue that we should improve this if possible.

Reply:
I think vim has word wrap on by default, so you only see it after executing ``set wrap!``

![image](https://user-images.githubusercontent.com/7258858/157460691-caaf9991-7460-45b3-9b37-bbe7c4e5eb19.png)